src/commons/navigationBar/NavigationBar.tsx

```diff
@@ -42,10 +42,52 @@ export type NavbarEntryInfo = {
   hiddenInBreakpoints?: ('xs' | 'sm' | 'md' | 'lg')[]; // hide text in Blueprint breakpoints
 };
 
-type CreateNavlinkFunction = (navbarEntry: NavbarEntryInfo) => React.ReactElement;
+const MobileHamburger: React.FC<{ navlinks: NavbarEntryInfo[] }> = ({ navlinks }) => {
+  // Don't render drawer when there are 0 navlinks in it
+  const [mobileSideMenuOpen, setMobileSideMenuOpen] = useState(false);
+  const shownNavlinks = navlinks.filter(e => !e.disabled);
+  const renderDrawer = shownNavlinks.length > 0;
+
+  const { courseShortName, courseId } = useSession();
+
+  return (
+    <NavbarGroup align={Alignment.LEFT}>
+      {renderDrawer && (
+        <Button
+          onClick={() => setMobileSideMenuOpen(!mobileSideMenuOpen)}
+          icon={IconNames.MENU}
+          large={true}
+          minimal={true}
+        />
+      )}
+      <NavLink
+        className={classNames('NavigationBar__link', Classes.BUTTON, Classes.MINIMAL)}
```

Comment:
Can we refactor this to use the new Button component as well? (since we are trying to move away from CSS)

Reply:
Resolved in c1c8532fc4ad887900e7c7c87df77114fcaea20d.